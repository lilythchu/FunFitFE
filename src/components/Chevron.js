import React from 'react';
import {TouchableOpacity} from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import globalStyles from '../../styles/global';

const Chevron = ({navigation, color}) => {
  return (
    <TouchableOpacity
      style={globalStyles.backIcon}
      onPress={() => navigation.goBack()}>
      <Entypo name="chevron-left" size={32} color={color} />
    </TouchableOpacity>
  );
};

export default Chevron;
