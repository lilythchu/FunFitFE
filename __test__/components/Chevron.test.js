import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import Chevron from '../../src/components/Chevron';

const navigation = {
  goBack: jest.fn()
};

describe('Chevron', () => {
  test('renders chevron correctly', () => {
    render(<Chevron />);
    expect(screen).toMatchSnapshot();
  });

  test('navigate correctly', () => {
    const {getByTestId} = render(<Chevron  navigation={navigation} />)
    fireEvent.press(getByTestId('chevron'));
    expect(navigation.goBack).toHaveBeenCalled();
  })
});