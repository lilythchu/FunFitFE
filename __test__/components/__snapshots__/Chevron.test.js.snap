// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chevron renders chevron correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 10,
      "marginLeft": 20,
      "marginTop": 40,
      "opacity": 1,
    }
  }
  testID="chevron"
>
  <Text />
</View>
`;
